transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -846,7 +992,16 @@ private void invokeWriteAndFlush(Object msg, ChannelPromise promise) {
     }
 
     private void write(Object msg, boolean flush, ChannelPromise promise) {
-        AbstractChannelHandlerContext next = findContextOutbound();
+        final AbstractChannelHandlerContext next;
```

Comment:
Can we somehow cache the result here?

Reply:
Sorry but what you want to cache ? We already cache the bitmask.

Reply:
I mean cache the result of this scan. The masks do not change, and the chain can change in theory, but usually doesn't. So the result of the loop is mostly the same, no?

Reply:
@nitsanw yeah it can change so I think we should not for now. That said we can easily do these kind of things once I will do https://github.com/netty/netty/issues/8511 as then we can remove all the volatile and can be sure all the changes are done in the EventLoop. In this case we can then just recalculate these things whenever the pipeline is mutated.

Reply:
A volatile load will be cheaper than recalc. And since adding and removing handlers already has some exclusion/order guarantees(I would think) we just need to ride on that.

Reply:
I am not sure about this and would like to consider thousand a follow up (as we did not cache before as well). Another downside is object size. As there are a lot of context instances we would like to keep the memory overhead as low as possible.